### ISSUE TITLE
Parenthesis mismatch in generated links

### ISSUE THREAD

**Issue by lehins** [haskell/haddock#1122]
Here is an example of a link that demonstrates the problem: [Luma](https://en.wikipedia.org/wiki/Luma_(video)). As you can see github's markdown correctly renders it, while haddock converts the same syntax:
```
[Luma](https://en.wikipedia.org/wiki/Luma_(video))
```
To a link:
```
<p><a href="https://en.wikipedia.org/wiki/Luma_(video">Luma</a>)</p>
```

The workaround of escaping the first closing parenthesis does the job, nevertheless it seems like it should not be a required step:
```
[Luma](https://en.wikipedia.org/wiki/Luma_(video\))
```

**Comment by harpocrates** [comment]
I agree that this is sub-optimal and would accept a fix. Commonmark says [this](https://spec.commonmark.org/0.29/#link-destination):

> A link destination consists of ...
>
>  *   a nonempty sequence of characters that does not start with `<`, does not include ASCII space or control characters, and includes parentheses only if (a) they are backslash-escaped or (b) they are part of a balanced pair of unescaped parentheses. (Implementations may impose limits on parentheses nesting to avoid performance issues, but at least three levels of nesting should be supported.)

Worth noting that Haddock's markup does not need to match markdown/commonmark, so this is really a feature request.
